MorseDecodeEncode/src/app/decode/decode.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { MorseCodeService } from '../morse-code.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DecodeService {
+  constructor(private morseCordeService: MorseCodeService) {}
+
+  public decode(morseCodeInput: string) {
+    let result = '';
+    let morseCodeLetter = '';
+
+    // for (let char of morseCodeInput) {
+    //   if(char === ' '){
+    //     result += this.morseCordeService.getLetterByMorseCode(morseCodeLetter);
+
+    //     morseCodeLetter = '';
+    //   }else if(char === '.' || char === '/' || char === '-'){
+    //     morseCodeLetter += char;
+    //   }
+    // }
+    // result += this.morseCordeService.getLetterByMorseCode(morseCodeLetter);
+
+    for (let i = 0; i < morseCodeInput.length; i++) {
+      let char = morseCodeInput[i];
+      if (char === ' ') {
```

Comment:
Possible, but splitting input (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split) by ` / ` and ` ` would have been simpler.